[mm_logs] follow up to add count info based on shape for inductor `aten.mm`s

YUNQIUGUO opened the pull request:
Summary:
as title.

when enable `TORCH_LOGS="+inductor"`, you can get logs at the end such as

stats [('calls_captured', 1), ('unique_graphs', 1)]
inductor [('pattern_matcher_count', 2), ('pattern_matcher_nodes', 2), ('benchmarking.TritonBenchmarker.benchmark_gpu', 2), **(('aten_addmm', (16, 6, 16)), 1)**, ('extern_calls', 1), ('async_compile_cache_miss', 1)]
graph_break []

Test Plan: follow up to add proper logging test.

Differential Revision: D70665104


cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

henrylhtsang commented on torch/_inductor/kernel/bmm.py:
```diff
@@ -168,6 +169,8 @@ def may_require_contiguous(t, meta_t):
     m, n, k, layout, mat1, mat2 = mm_args(mat1, mat2, layout=layout)
 
+    # below is for getting an overview logging info of inductor mms
+    counters["inductor"][("aten.bmm",(m, n, k))] = counters["inductor"].get(("aten.bmm",(m, n, k)), 0) + 1
```
I suspect you don't need get, since its a lot of defaultdict.

YUNQIUGUO committed:
[mm_logs] follow up to add count info based on shape for inductor `aten.mm`s (#148623)

Summary:

as title.

when enable `TORCH_LOGS="+inductor"`, you can get logs at the end such as


stats [('calls_captured', 1), ('unique_graphs', 1)]
inductor [('pattern_matcher_count', 2), ('pattern_matcher_nodes', 2), ('benchmarking.TritonBenchmarker.benchmark_gpu', 2), **(('aten_addmm', (16, 6, 16)), 1)**, ('extern_calls', 1), ('async_compile_cache_miss', 1)]
graph_break []

Test Plan: follow up to add proper logging test.

Reviewed By: henrylhtsang

Differential Revision: D70665104
